websites/management/commands/markdown_cleaning/rules/broken_link_fix.py

```diff
@@ -0,0 +1,284 @@
+                    )
+                    if content:
+                        matched_contents.append(content)
+                except KeyError:
+                    pass
+        if len(matched_contents) == 1:
+            return matched_contents[0]
+
+        return None
+
+    def _find_course_url_replacement(
+        self, result: LinkParseResult, url_path: str
+    ) -> str | None:
+        """
+        Return an absolute course url to use as a replacement
+        if `url_path` is a relative and valid course path.
+        """
+        try:
+            website_by_path = self.content_lookup.find_website_by_url_path(url_path)
+        except KeyError:
+            website_by_path = None
+
+        if website_by_path and website_by_path.unpublish_status is None:
+            # This is a course URL.
+            # These need to be absolute URLs to work correctly.
+            link = dataclasses.replace(result.link)
+            link.destination = "/" + link.destination
+            return link.to_markdown()
+
+        return None
+
+    def _find_content_and_replacement(
+        self, result: LinkParseResult, wc: WebsiteContent, url: ParseResult
+    ) -> (str, ReplacementNotes):
+        """
+        Find the content that best matches the `url` and return the
+        replacement.
+        """
+        Notes = partial(
+            self.ReplacementNotes, issue_type=None, linked_content=None, fix=None
+        )
+
+        url_path = url.path.rstrip("/") or "/"
+        if not url_path.startswith("/"):
+            # Most likely a relative URL like "pages/syllabus".
+            # We'll make it root-relative to help with locating content.
+            url_path = f"{get_rootrelative_url_from_content(wc)}/{url_path}"
+
+        try:
+            found_wc = self.content_lookup.find(url_path, base_site=wc.website)
+        except KeyError:
+            found_wc = None
+
+        replacement, notes = result.original_text, Notes()
+
+        if found_wc and url_path.endswith(("_index", "index.htm", "index.html")):
+            # Remove the extra _index* url component.
+            replacement, fix_note = self.create_replacement(result, url, found_wc)
+            notes = Notes(
+                issue_type="_index postfix",
+                fix=fix_note,
+                linked_content=found_wc.text_id,
+            )
+        elif found_wc:
+            # A WebsiteContent was successfully discovered. We probably don't
+            # need to do anything.
+            replacement, notes = result.original_text, Notes(issue_type="Nothing to do")
+        else:
+            # Start making guesses.
+            found_wc = self._find_best_matching_content(url, wc)
+            notes = Notes(issue_type="Unknown link")
+
+            if found_wc:
+                replacement, fix_note = self.create_replacement(result, url, found_wc)
+                notes = Notes(
+                    issue_type="Unknown link",
```

Comment:
```suggestion
            notes = Notes(issue_type="Found unique fuzzy match")
```
or something?  To distinguish from the Note just above.